app/api/speakers.py

```diff
@@ -45,6 +46,12 @@ def before_post(self, args, kwargs, data=None):
                                                          deleted_at=None)) > 0:
             raise ForbiddenException({'pointer': ''}, 'Speaker with this Email ID already exists')
 
+        if not has_access('is_coorganizer', event_id=data['event']) and data.get('is_email_overriden'):
+            raise ForbiddenException({'pointer': 'data/attributes/is_email_overriden'},
+                                     'Co-organizer access required to override email')
+        elif has_access('is_coorganizer', event_id=data['event']) and data.get('is_email_overriden'):
+            data['email'] = current_user.email
```

Comment:
If email is overridden, then when is this email used?

Reply:
If email is overriden then organizer's email is used. I'm adding this line so that we don't depend solely on client to send email.

Reply:
Yeah, but in above functions, you don't send email if it is overriden. Reason?

Reply:
Oh I misread your first question. This email acts as a dummy to satisfy the schema constraints (as email cannot be null) and can have secondary use to let us know which co-organizer added him/her.
And we are not sending email because it will act as a spam to organizer since he is already getting every notif and email.

Reply:
Then only change data['email'] if it is not present. Or else, it is wrong info. Also, if someone disables email override, you will now definitely spam the organizer

Reply:
Updated the logic and updated data['email'] only if it isn't present